models/actions/artifact.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// This artifact server is inspired by https://github.com/nektos/act/blob/master/pkg/artifacts/server.go.
+// It updates url setting and uses ObjectStore to handle artifacts persistence.
+
+package actions
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
+)
+
+const (
+	// ArtifactUploadStatusPending is the status of an artifact upload that is pending
+	ArtifactUploadStatusPending = 1
+	// ArtifactUploadStatusConfirmed is the status of an artifact upload that is confirmed
+	ArtifactUploadStatusConfirmed = 2
+)
+
+func init() {
+	db.RegisterModel(new(ActionArtifact))
+}
+
+// ActionArtifact is a file that is stored in the artifact storage.
+type ActionArtifact struct {
+	ID               int64 `xorm:"pk autoincr"`
+	RunID            int64 `xorm:"index unique(run-id-name)"` // The run id of the artifact
+	RunnerID         int64
+	RepoID           int64 `xorm:"index"`
+	OwnerID          int64
+	CommitSHA        string
+	StoragePath      string             // The path to the artifact in the storage
+	FileSize         int64              // The size of the artifact in bytes
+	FileGzipSize     int64              // The size of the artifact in bytes after gzip compression
+	ContentEncnoding string             // The content encoding of the artifact
+	ArtifactPath     string             // The path to the artifact when runner uploads it
+	ArtifactName     string             `xorm:"unique(run-id-name)"` // The name of the artifact when runner uploads it
+	UploadStatus     int64              `xorm:"index"`               // The status of the artifact upload
+	Created          timeutil.TimeStamp `xorm:"created"`
+	Updated          timeutil.TimeStamp `xorm:"updated index"`
+}
+
+// CreateArtifact create a new artifact with task info or get same named artifact in the same run
+func CreateArtifact(ctx context.Context, t *ActionTask, artifactName string) (*ActionArtifact, error) {
+	if t.Job == nil {
+		if err := t.LoadJob(ctx); err != nil {
+			return nil, err
+		}
+	}
+	if err := t.Job.LoadRun(ctx); err != nil {
+		return nil, err
+	}
+	artifact, _ := GetArtifactByArtifactName(ctx, t.Job.RunID, artifactName)
```

Comment:
Why ignore the error? I think only artifact does not exist error could be ignored, and others should not be ignored?